matrix-sdk/src/main/java/org/matrix/androidsdk/data/store/MXFileStore.java

```diff
@@ -2293,13 +2296,20 @@ private boolean writeObject(String description, File file, Object object) {
         boolean succeed = false;
         try {
             FileOutputStream fos = new FileOutputStream(file);
-            GZIPOutputStream gz = CompatUtil.createGzipOutputStream(fos);
-            ObjectOutputStream out = new ObjectOutputStream(gz);
+            CipherOutputStream cos = CompatUtil.createCipherOutputStream(fos, mContext);
+
+            if (cos != null) {
+                GZIPOutputStream gz = CompatUtil.createGzipOutputStream(cos);
+                ObjectOutputStream out = new ObjectOutputStream(gz);
 
-            out.writeObject(object);
-            out.close();
+                out.writeObject(object);
+                out.flush();
+                out.close();
 
-            succeed = true;
+                succeed = true;
+            } else {
+                Log.e(LOG_TAG, "## writeObject() failed to open cipher output stream");
```

Comment:
You should write object unencrypted for compatibility mode

Reply:
Are you sure this is a good idea ?

Reply:
If the SDK is run on a Android 4.1, I think the cache will not work, no?